src/com/maddyhome/idea/vim/group/SearchGroup.java

```diff
@@ -625,20 +617,17 @@ public static TextRange findNext(@NotNull Editor editor, @NotNull String pattern
       return null;
     }
     final int size = EditorHelper.getFileSize(editor);
-    final TextRange max = Collections.max(results, new Comparator<TextRange>() {
-      @Override
-      public int compare(TextRange r1, TextRange r2) {
-        final int d1 = distance(r1, offset, forwards, size);
-        final int d2 = distance(r2, offset, forwards, size);
-        if (d1 < 0 && d2 >= 0) {
-          return Integer.MAX_VALUE;
-        }
-        return d2 - d1;
+    final TextRange max = Collections.max(results, (r1, r2) -> {
+      final int d1 = distance(r1, offset, forwards, size);
+      final int d2 = distance(r2, offset, forwards, size);
+      if (d1 < 0 && d2 >= 0) {
+        return Integer.MAX_VALUE;
       }
+      return d2 - d1;
     });
     if (!Options.getInstance().isSet("wrapscan")) {
       final int start = max.getStartOffset();
-      if (forwards && start < offset || start >= offset) {
+      if (forwards || start >= offset) {
```

Comment:
This refactoring looks unclear. Let's keep the old version or split it into two ifs.